### PR TITLE
LG-3814 Fixes for LN TrueID failing the doc_pii_form validation check

### PR DESCRIPTION
**Pull request by amathews-fs**
Adding modifications to LN TrueID to assure the correct pii is returned for the rest of the verification process to proceed as expected.

### CODE REVIEW

File: lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb [hunk 1]
```diff
@@ -36,10 +69,10 @@ def error_messages
         end
 
         def extra_attributes
-          if true_id_product.present? && true_id_product[:AUTHENTICATION_RESULT].present?
+          if true_id_product&.dig(:AUTHENTICATION_RESULT).present? && true_id_product[:AUTHENTICATION_RESULT].present?
```
Comment (solipet) [inline lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb]: Don't need the right side of the `&&`

File: lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb [hunk 2]
```diff
@@ -54,11 +87,16 @@ def extra_attributes
         end
 
         def pii_from_doc
-          return {} unless true_id_product&.dig(:AUTHENTICATION_RESULT).present?
+          return {} unless true_id_product&.dig(:IDAUTH_FIELD_DATA).present?
 
-          true_id_product[:AUTHENTICATION_RESULT].select do |k, _v|
-            PII_DETAILS.include? k
+          pii = {}
+          PII_INCLUDES.each do |key, value|
```
Comment (solipet) [inline lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb]: I might rename these `|ln_key, local_key|` - it confused me for a sec that you were using the "value" as the key.

**Review by solipet — APPROVED** [review]
Looks good - just a couple of comments.

File: lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb (hunk 2 above)
Comment (amathews-fs) [reply lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb]: Good call.

File: lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb (hunk 1 above)
Comment (amathews-fs) [reply lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb]: Oops! That was supposed to have collapsed.

Comment (amathews-fs) [reply lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb]: Not totally sure how I managed that...

File: lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb [hunk 3]
```diff
@@ -54,11 +87,16 @@ def extra_attributes
...
+            pii[value] = true_id_product[:IDAUTH_FIELD_DATA][key]
```
Comment (zachmargolis) [inline lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb]: Maybe call `value` something like `mapped_key` or something to make it clearer, it's not a value?

```suggestion
          PII_INCLUDES.each do |key, mapped_key|
            pii[mapped_key] = true_id_product[:IDAUTH_FIELD_DATA][key]
```

File: spec/identity_doc_auth/lexis_nexis/responses/true_id_response_spec.rb [hunk 4]
```diff
@@ -54,7 +54,7 @@
     end
     it 'has PII data' do
       pii_from_doc = described_class.new(success_response, false, config).pii_from_doc
-      expect(pii_from_doc).not_to be_empty
+      expect(pii_from_doc).to include(:first_name, :last_name, :dob, :state)
```
Comment (zachmargolis) [inline spec/identity_doc_auth/lexis_nexis/responses/true_id_response_spec.rb]: can we check the values here? Especially useful for checking the DOB formatting code

**Review by zachmargolis — APPROVED** [review]
LGTM

File: lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb (hunk 3 above)
Comment (amathews-fs) [reply lib/identity_doc_auth/lexis_nexis/responses/true_id_response.rb]: Yeah, I changed it to be more expressive on both parameters.

File: spec/identity_doc_auth/lexis_nexis/responses/true_id_response_spec.rb (hunk 4 above)
Comment (amathews-fs) [reply spec/identity_doc_auth/lexis_nexis/responses/true_id_response_spec.rb]: Done.

**Pull request merged**